src/Mailer/MailSentCommandHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Mailer;
+
+use Broadway\CommandHandling\CommandHandler;
+use CultuurNet\UDB3\CommandHandling\AsyncCommand;
+use CultuurNet\UDB3\CommandHandling\AsyncCommandTrait;
+use CultuurNet\UDB3\Mailer\Event\SentMail;
+use DateTimeImmutable;
+use Psr\Log\LoggerInterface;
+
+class MailSentCommandHandler implements CommandHandler, AsyncCommand
+{
+    use AsyncCommandTrait;
+
+    private Mailer $mailer;
+    private MailsSentRepository $mailsSentRepository;
+    private LoggerInterface $logger;
+
+    public function __construct(Mailer $mailer, MailsSentRepository $mailsSentRepository, LoggerInterface $logger)
+    {
+        $this->mailer = $mailer;
+        $this->mailsSentRepository = $mailsSentRepository;
+        $this->logger = $logger;
+    }
+
+    public function handle($command): void
+    {
+        if (!$command instanceof SentMail) {
```

Comment:
This shouldn't be necessary when you add a typehint to the `handle()`. If I look at `EventExportCommandHandler` it looks like the commanbus is auto-wired and selects its commandhandler based on the typehints.

Reply:
The refactored handler will handle 3 types of Commands later on.